docs/releases/upgrading/upgrade-v12.0.md

```diff
@@ -0,0 +1,164 @@
+# Upgrading from v11 to v12
+
+## Prerequisites
+
+The steps listed in this article require an existing local installation of InvenioRDM v11.
+
+!!! warning "Backup"
+
+    Always backup your database and files before you try to perform an upgrade.
+
+!!! info "Older Versions"
+
+    In case you have an InvenioRDM installation older than v11, you can gradually upgrade
+    using the existing infrastructure to v11 and afterwards continue from here.
+
+## Upgrade Steps
+
+!!! warning "Upgrade your invenio-cli"
+
+    Make sure you have the latest `invenio-cli` installed. For InvenioRDM v12 it is v1.2.0
+
+    ```bash
+    $ invenio-cli --version
+    invenio-cli, version 1.
+    ```
+
+!!! info "Virtual environments"
+
+    In case you are not inside a virtual environment, make sure that you prefix each `invenio`
+    command with `pipenv run`.
+
+
+**Local development**
+
+Changing the Python version in your development environment highly
+depends on your setup, and there is no golden rule.
+One example would be to use [PyEnv](https://github.com/pyenv/pyenv).
+
+You should delete your virtualenv before running `invenio-cli` or `pipenv` commands below.
+
+!!! warning "Risk of losing data"
+
+    Your virtual env folder contains uploaded files in InvenioRDM, in `var/instance/data`.
+    If you need to keep such files, make sure you copy them over to the new virtual env in the same location.
+
+### Upgrade InvenioRDM
+
+(TODO: should we write that 3.9<=X<=3.12 is possible?)
+Make sure that your virtual env is now running with Python 3.9.
+
+Upgrade the RDM version:
+
+```bash
+cd <my-site>
+# Upgrade to InvenioRDM v12
+invenio-cli packages update 12.0.0
+pipenv uninstall flask-babelex
+# Re-build assets
+invenio-cli assets build
+```
+
+Optionally, update the file `<my-site>/Pipfile`. Attention: this action might lead to
+installing unwanted pre-releases of other packages.
+
+```diff
+[packages]
+---invenio-app-rdm = {extras = [...], version = "~=11.0.0"}
++++invenio-app-rdm = {extras = [...], version = "~=12.0.0"}
+
```

Comment:
This change was a breaking change in pipenv package. Should we include it in the upgrade recipe?

```diff
- my-site = {editable="True", path="./site"}
+ my-site = {editable=true, path="./site"}
```

Reply:
Yeah, it would be good to add, even though it's not technically a v12 change